library/database/class.sqldriver.php

```diff
@@ -1427,30 +1428,67 @@ public function options($key, $value = null) {
         } elseif ($value !== null) {
             $this->_Options[$key] = $value;
             return $this;
-        } elseif (isset($this->_Options[$key]))
+        } elseif (isset($this->_Options[$key])) {
             return $this->_Options[$key];
-        else {
+        } else {
             return null;
         }
     }
 
     /**
      * Adds to the $this->_OrderBys collection.
      *
-     * @param string $fields A string of fields to be ordered.
+     * This method supports many different formats for legacy reasons.
+     *
+     * The current best practice is:
+     *
+     * ```
+     * $sql->orderBy(['column1', '-column2', ...]); // prefix with "-" for descending
+     * $sql->orderBy('column1')
+     *     ->orderBy('-column2')
+     * ```
+     *
+     * Legacy versions include:
+     *
+     * ```
+     * $sql->orderBy('column', 'asc|desc');
+     * $sql->orderBy('column1 desc, column2', '');
+     * $sql->orderByy(['column1' => 'desc', 'column2' => 'asc']);
+     * ```
+     *
+     * @param string|array|null $fields A string of fields to be ordered.
      * @param string $direction The direction of the sort.
      * @return Gdn_SQLDriver $this
      */
     public function orderBy($fields, $direction = 'asc') {
+        $r = $this->orderByPrefix('', $fields, $direction);
+        return $r;
+    }
+
+    /**
+     * Order by a field or fields, adding a prefix to each fields.
+     *
+     * @param string $px
+     * @param string|array|null $fields
+     * @param string $direction
+     * @return $this
+     * @see Gdn_SQLDriver::orderBy()
+     */
+    public function orderByPrefix(string $px, $fields, $direction = 'asc') {
         if (!$fields) {
```

Comment:
This little refactor in `GDN_SQLDriver` could have been its own PR, but basically I was finding it to be such a hassle wrangling all of our various sorting methods that I decided to make it easier on myself and add support for them here.